.circleci/config.yml

```diff
@@ -1,977 +1,67 @@
-          requires:
-            - generate-sql
-            # Public image must be pushed after the private one because of
-            # webhooks used in Ops logic. For details, see:
-            # https://bugzilla.mozilla.org/show_bug.cgi?id=1715628#c0
-            - deploy-to-private-gcr
-          filters:
-            branches:
-              only:
-                - main
-      # The following "private" jobs are basically clones of the public jobs
-      # for generate-sql, deploy, and push-generated-sql, except that they pull
-      # in some additional content from an internal Mozilla repository for
-      # cases where ETL code cannot be public. Although the CI logic is
-      # consolidated in this public repository, note that we are both pulling
-      # from the internal repository and pushing generated results back to
-      # a branch on that internal repository, which may be initially
-      # surprising.
-      - private-generate-sql:
-          filters:
-            branches:
-              only:
-                - main
-      - push-private-generated-sql:
-          requires:
-            - private-generate-sql
-          filters:
-            branches:
-              only:
-                - main
-      - sync-dags-repo:
-          name: 🔃 Synchronize private-bigquery-etl submodule
-          repo-to-sync: private-bigquery-etl
-          target-branch: private-generated-sql
-          requires:
-            - push-private-generated-sql
-          filters:
-            branches:
-              only:
-                - main
-      - deploy-to-private-gcr:
-          context: data-eng-airflow-gcr
-          requires:
-            - private-generate-sql
-            # can't run in parallel because CIRCLE_BUILD_NUM is same
-            - build
-            - generate-sql
-          filters:
-            branches:
-              only:
-                - main
-  tagged-deploy:
+  filter-paths-main:
+    when:
+      equal: [main, << pipeline.git.branch >>]
+    jobs:
+      - path-filtering/filter:
+          base-revision: << pipeline.git.base_revision >>
+          config-path: .circleci/workflows.yml
+          mapping: |
+            .* validate-bqetl true
+            .* validate-routines true
+            .* validate-sql true
+            .* deploy true
+  filter-paths-pr:
+    when:
+      and:
+        - not:
+            equal: [main, << pipeline.git.branch >>]
     jobs:
-      - deploy-to-pypi:
-          filters:
-            tags:
-              only: /[0-9]{4}.[0-9]{1,2}.[0-9]+/  # Calver: YYYY.M.MINOR
-            branches:
-              # Ignore all branches; this workflow should only run for tags.
-              ignore: /.*/
-  nightly:
-    # Run after schema-generator to ensure we are up-to-date
-    triggers:
-      - schedule:
-          cron: "0 5 * * *"
-          filters:
-            branches:
-              only:
-                - main
-    jobs: *build_jobs
```

Comment:
This is removing the `nightly` workflow without a replacement.  I'm guessing that's intentional since that workflow hasn't really been working as intended, but it'd be good to note (e.g. in the PR description).